src/lib/util/FileHandler.ts

```diff
@@ -77,4 +79,16 @@ export class FileHandler {
 			});
 		});
 	}
+
+	tmpFile(options: object = {}): Promise<FileResult> {
```

Comment:
Good idea, at first I thought FileHandler was delegating to methods without adding any abstraction, but I see that it isn't.

The `tmp` docs aren't great, the `keep` option is ignored unless you call another method, but it's worth having our new method set this up and not leave it up to the caller.

I'm nor sure I understand "You might as well make this async". The async keyword allows methods to use the `await` keyword inside of them or return simple objects that are converted to promises automatically. Is there any benefit since the code isn't using `await` and are returning a Promise? If, so, should we convert all methods in that file to use the async keyword?